dev/hooks/pre-commit

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Run rustfmt on the whole repository using nightly toolchain
+cargo +nightly fmt --all -- --check
```

Comment:
why nightly?

Reply:
because it has the features for formatting imports which stable does not have